apollo-compiler/src/test/graphql/com/example/antlr_tokens/adapter/TestQuery_ResponseAdapter.kt

```diff
@@ -5,120 +5,130 @@
+        else -> break
       }
-      TestQuery.Data(
-        typeWithGraphQLKeywords = typeWithGraphQLKeywords
-      )
     }
+    reader.endObject()
+    return TestQuery.Data(
+      typeWithGraphQLKeywords = typeWithGraphQLKeywords
+    )
   }
 
-  override fun toResponse(writer: ResponseWriter, value: TestQuery.Data) {
-    if(value.typeWithGraphQLKeywords == null) {
-      writer.writeObject(RESPONSE_FIELDS[0], null)
-    } else {
-      writer.writeObject(RESPONSE_FIELDS[0]) { writer ->
-        TypeWithGraphQLKeywords.toResponse(writer, value.typeWithGraphQLKeywords)
-      }
-    }
+  override fun toResponse(writer: JsonWriter, value: TestQuery.Data) {
+    writer.beginObject()
+    writer.name("typeWithGraphQLKeywords")
+    nullableTypeWithGraphQLKeywordsAdapter.toResponse(writer, value.typeWithGraphQLKeywords)
+    writer.endObject()
   }
 
-  object TypeWithGraphQLKeywords : ResponseAdapter<TestQuery.Data.TypeWithGraphQLKeywords> {
+  companion object {
     val RESPONSE_FIELDS: Array<ResponseField> = arrayOf(
       ResponseField(
-        type = ResponseField.Type.Named.Other("String"),
-        responseName = "on",
-        fieldName = "on",
-        arguments = emptyMap(),
-        conditions = emptyList(),
-        fieldSets = emptyList(),
-      ),
-      ResponseField(
-        type = ResponseField.Type.Named.Other("String"),
-        responseName = "null",
-        fieldName = "null",
-        arguments = mapOf<String, Any?>(
-          "fragment" to mapOf<String, Any?>(
-            "kind" to "Variable",
-            "variableName" to "operation")),
-        conditions = emptyList(),
-        fieldSets = emptyList(),
-      ),
-      ResponseField(
-        type = ResponseField.Type.Named.Other("String"),
-        responseName = "alias",
-        fieldName = "null",
-        arguments = mapOf<String, Any?>(
-          "fragment" to """
-          |A string
-          |with a new line
-          """.trimMargin()),
-        conditions = emptyList(),
-        fieldSets = emptyList(),
+        type = ResponseField.Type.Named.Object("TypeWithGraphQLKeywords"),
+        fieldName = "typeWithGraphQLKeywords",
+        fieldSets = listOf(
+          ResponseField.FieldSet(null, TypeWithGraphQLKeywords.RESPONSE_FIELDS)
+        ),
       )
     )
 
-    override fun fromResponse(reader: ResponseReader, __typename: String?):
-        TestQuery.Data.TypeWithGraphQLKeywords {
-      return reader.run {
-        var on: String? = null
-        var null_: String? = null
-        var alias: String? = null
-        while(true) {
-          when (selectField(RESPONSE_FIELDS)) {
-            0 -> on = readString(RESPONSE_FIELDS[0])
-            1 -> null_ = readString(RESPONSE_FIELDS[1])
-            2 -> alias = readString(RESPONSE_FIELDS[2])
-            else -> break
-          }
+    val RESPONSE_NAMES: List<String> = RESPONSE_FIELDS.map { it.responseName }
+  }
+
+  class TypeWithGraphQLKeywords(
```

Comment:
maybe also all nested classed make private?

Reply:
Yep, good idea. It doesn't quite work at the moment because `ResponseFields` "jump" class nesting:

```
class Data_Adapter {
  companion object RESPONSE_FIELDS = arrayOf(ResponseField(fieldSets = listOf(
    // Cannot access 'HumanCharacterHero': it is private in 'Hero'`
    FieldSet("Human", Hero.HumanCharacterHero.RESPONSE_FIELDS)
    FieldSet(null, Hero.OtherCharacterHero.RESPONSE_FIELDS)
  ))

    private class Hero {
       private class HumanCharacterHero {...}
       private class OtherCharacterHero {...}
    }  
}
```